### PR TITLE
chore: allow timeout to be set

### PR DESCRIPTION
**Pull request by RadekMolenda**
## 🐻 What your PR is doing?

Allow setting timeout in BearerClient

## 📦 Package concerned

@bearer/node

### CODE REVIEW

File: packages/node/src/client.ts [hunk 1]
```diff
@@ -88,7 +92,7 @@ class BearerClient {
 
     const preheaders: BearerHeaders = {
       Authorization: this.bearerApiKey,
-      'User-Agent': 'Bearer.sh',
+      'User-Agent': 'Bearer-Node',
```
Comment (cfabianski) [inline packages/node/src/client.ts]: Version?

Comment (RadekMolenda) [reply packages/node/src/client.ts]: done

File: packages/node/src/client.ts [hunk 2]
```diff
@@ -36,7 +38,9 @@ class BearerClient {
     readonly bearerApiKey: string,
     readonly setupId?: string,
     readonly authId?: string
-  ) {}
+  ) {
+    this.client.defaults.timeout = this.options.timeout
```
Comment (tanguyantoine) [inline packages/node/src/client.ts]: here, you are updating axios defaults, you can impact other libraries that use axios

Comment (tanguyantoine) [reply packages/node/src/client.ts]: Alternatively you can pass the timeout to each request: https://github.com/Bearer/bearer/pull/747/files#diff-a00e1d800e81b35855676e786345f4e9L115
https://github.com/Bearer/bearer/pull/747/files#diff-a00e1d800e81b35855676e786345f4e9L133

Comment (RadekMolenda) [reply packages/node/src/client.ts]: I feel uncomfortable setting the timeout per request - it doesn't seem to be dry enough. Have a look at current implementation pls. WDYT? I also updated tests to ensure headers are set correctly

Comment (tanguyantoine) [reply packages/node/src/client.ts]: fair enough 👍

File: packages/node/README.md [hunk 3]
```diff
@@ -67,6 +67,23 @@ github
...
 ```
 
+### Setting the request timeout
+
+By default bearer client times out after 5 seconds. Bearer allows to increase the timeout to up to 30 seconds
+
+``` tsx
+import bearer from '@bearer/node'
+
+const client = bearer(process.env.BEARER_SECRET_KEY, { timeout: 10 * 1000 }) // sets the timeout to 10 seconds
```
Comment (tanguyantoine) [inline packages/node/README.md]: I'm wondering if we could use seconds rather than ms, what do you think?

Comment (RadekMolenda) [reply packages/node/README.md]: don't know - thought node and javascript speaks in ms

**Comment by RadekMolenda** [comment]
https://github.com/Bearer/bug-tracker/issues/3

**Pull request merged**